kivy/uix/carousel.py

```diff
@@ -446,12 +450,19 @@ def on_index(self, *args):
         self._trigger_position_visible_slides()
         self._offset = 0
 
+    def on_loop(self, *args):
```

Comment:
It is generally better to directly `fbind` `_insert_visible_slides` to `loop` in the `__init__` method, than to use the `on_loop` method. I know this principle is frequently violated in the Kivy codebase (even in `on_slides` below), but I try to fix it when I happen to work on code that does this.

The problem with this pattern is that if a user inherits and overwrites the `on_xxx` method for their own reason, we lose all the stuff that we wanted to happen as a result to `xxx` changing. So it's better to bind directly. For this case you'd need to bind a local function that eats the `args` passed from the callback, but it's still better.

Reply:
Yes, I understand what you mean. I'll fix that.